modules/dnn/src/op_webnn.hpp

```diff
@@ -0,0 +1,166 @@
+
+    InferenceEngine::ExecutableNetwork netExec;
+    InferenceEngine::BlobMap allBlobs;
+    std::string device_name;
+    bool isInit = false;
+
+    struct NgraphReqWrapper
+    {
+        NgraphReqWrapper() : isReady(true) {}
+
+        void makePromises(const std::vector<Ptr<BackendWrapper> >& outs);
+
+        InferenceEngine::InferRequest req;
+        std::vector<cv::AsyncPromise> outProms;
+        std::vector<std::string> outsNames;
+        bool isReady;
+    };
+    std::vector<Ptr<NgraphReqWrapper> > infRequests;
+
+    InferenceEngine::CNNNetwork cnn;
+    bool hasNetOwner;
+    std::vector<std::string> requestedOutputs;
+    std::unordered_set<std::shared_ptr<ngraph::Node>> unconnectedNodes;
+};
+
+class WebNNBackendNode : public BackendNode
+{
+public:
+    InfEngineNgraphNode(const std::vector<Ptr<BackendNode> >& nodes, Ptr<Layer>& layer,
+                        std::vector<Mat*>& inputs, std::vector<Mat>& outputs,
+                        std::vector<Mat>& internals);
+
+    InfEngineNgraphNode(std::shared_ptr<ngraph::Node>&& _node);
+    InfEngineNgraphNode(std::shared_ptr<ngraph::Node>& _node);
+
+    void setName(const std::string& name);
+
+    // Inference Engine network object that allows to obtain the outputs of this layer.
+    std::shared_ptr<ngraph::Node> node;
+    Ptr<InfEngineNgraphNet> net;
+    Ptr<dnn::Layer> cvLayer;
+};
+
+class WebNNBackendWrapper : public BackendWrapper
+{
+public:
+    NgraphBackendWrapper(int targetId, const Mat& m);
+    NgraphBackendWrapper(Ptr<BackendWrapper> wrapper);
+    ~NgraphBackendWrapper();
+
+    static Ptr<BackendWrapper> create(Ptr<BackendWrapper> wrapper);
+
+    virtual void copyToHost() CV_OVERRIDE;
+    virtual void setHostDirty() CV_OVERRIDE;
+
+    InferenceEngine::DataPtr dataPtr;
+    InferenceEngine::Blob::Ptr blob;
+    AsyncArray futureMat;
+};
+
+InferenceEngine::DataPtr ngraphDataNode(const Ptr<BackendWrapper>& ptr);
+
+// This is a fake class to run networks from Model Optimizer. Objects of that
+// class simulate responses of layers are imported by OpenCV and supported by
+// Inference Engine. The main difference is that they do not perform forward pass.
+class NgraphBackendLayer : public Layer
+{
+public:
+    NgraphBackendLayer(const InferenceEngine::CNNNetwork &t_net_) : t_net(t_net_) {};
+
+    virtual bool getMemoryShapes(const std::vector<MatShape> &inputs,
+                                 const int requiredOutputs,
+                                 std::vector<MatShape> &outputs,
+                                 std::vector<MatShape> &internals) const CV_OVERRIDE;
+
+    virtual void forward(InputArrayOfArrays inputs, OutputArrayOfArrays outputs,
+                         OutputArrayOfArrays internals) CV_OVERRIDE;
+
+    virtual bool supportBackend(int backendId) CV_OVERRIDE;
+
+private:
+    InferenceEngine::CNNNetwork t_net;
+};
+
+#endif  // HAVE_WebNN
+
+constexpr bool haveWebNN() {
```

Comment:
Please only keep this function and remove others. Please add others in your next PR.